feat: add base navigation bar

divyanshugarg36 opened the pull request:
- add roboto font
- add document.tsx for full web configuration
- add layout

divyanshugarg36 committed:
feat: add base navigation bar

- add roboto font
- add document.tsx for full web configuration
- add layout

theamanjs commented on src/components/Navigation.tsx:
```diff
@@ -0,0 +1,22 @@
+import Image from 'next/image';
+import Link from 'next/link';
+import React from 'react';
```
I believe you don't need to import React on every file while using next.js boilerplate. Can you remove it from here and all other files where you imported them?

divyanshugarg36 replied:
I have updated this one

theamanjs replied:
Can you link the changes in this thread? @divyanshugarg36

divyanshugarg36 replied:
This change is not reflected here due to the reason I am actually using React import for typescript specifications.
We might not need the react import for a component to work but typescript type specification is always good to have

divyanshugarg36 replied:
Apart from that i can assure that the react is only imported at the places where we are actually using the variable

theamanjs commented on src/styles/navigation.scss:
```diff
@@ -0,0 +1,40 @@
+nav {
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
```
I think you are probably gonna need to reuse this property in styling. I saw you have created `flex-center` mixin. You can create another mixin that could be something like:

```scss
@mixin flex(
  $justifyContent: space-between,
  $alignItems: center,
  $direction: row
) {
  display: flex;
  justify-content: $justifyContent;
  align-items: $alignItems;
  flex-direction: $direction;
}
```

divyanshugarg36 replied:
For both situations considering we create this mixing I am thinking about the problem, it can cause of having unused code for example if I just want to apply display flex and align-items center, In that situation, I will be adding 2 extra properties for every place I will be using flex and same for font group. Thus leading to larger file size for CSS maybe not by a huge margin but with some parts.
Views?...

theamanjs replied:
I don't think you will have to repeat the unwanted properties, you can always mentioned the specific argument. Just in case you want to add the align-items center property you can write something like
```scss
@include flex($align-items: center);
```

And it will pick the rest of the properties as default mentioned. Let  me know if I am wrong.

divyanshugarg36 replied:
Yeah, I understand your point but I am more concerned about post-compiled code.
For example, let's say I want to use flex and want to use only 2 of these 4 properties, 
If I am using this mixing there would be a situation where even for using 2 properties it will output 4 properties for the same even if we are mentioning default properties. So overall I just want to save extra lines in the compiled version even if it means a few extra lines in code but it will save overall performance not by a large margin but maybe small. We can look at other solutions for this

divyanshugarg36 replied:
https://dev.to/nicolalc/a-mixin-to-rule-them-all-flexbox-driven-layout-mixin-1h43
Found something
I will work on this one in a separate PR right next to the last one I just shared #13

divyanshugarg36 replied:
https://github.com/divyanshugarg36/brl-cad-web/pull/16
Fixes this

theamanjs replied:
Great job :raised_hands: Can you do something similar with the font properties as well? @divyanshugarg36

divyanshugarg36 replied:
#14  Issue has been created and PR will be sent soon

divyanshugarg36 committed:
chore: add mobile view for navigation bar

- add useoutside hook
- add navigation icon
- add navigation links

theamanjs commented on src/styles/navigation.scss:
```diff
@@ -0,0 +1,83 @@
+
+    h1 {
+      margin-left: $rhythm !important;
+      font-size: $rhythm * 2;
+      font-weight: bold;
+      color: var(--medium-grey);
+    }
```
similar to the above mentioned flex mixin you can create one for text too.

divyanshugarg36 replied:
#14 Issue has been created and PR will be sent soon

theamanjs replied:
COOOOOOOOOOOOOOL

divyanshugarg36 committed:
chore: fix navigation key issue

- remove react imports from unrequired places
- add interface for constant data
